Zotero/Scenes/Detail/PDF/Views/CustomFreeTextAnnotationView.swift

```diff
@@ -0,0 +1,186 @@
+    func showColorPicker(sender: UIView, key: PDFReaderState.AnnotationKey, updated: @escaping (String) -> Void)
+    func showFontSizePicker(sender: UIView, key: PDFReaderState.AnnotationKey, updated: @escaping (UInt) -> Void)
+    func showTagPicker(sender: UIView, key: PDFReaderState.AnnotationKey, updated: @escaping ([Tag]) -> Void)
+    func deleteAnnotation(sender: UIView, key: PDFReaderState.AnnotationKey)
+    func change(fontSize: UInt, for key: PDFReaderState.AnnotationKey)
+    func getColor(for key: PDFReaderState.AnnotationKey) -> UIColor?
+    func getFontSize(for key: PDFReaderState.AnnotationKey) -> UInt?
+    func getTags(for key: PDFReaderState.AnnotationKey) -> [Tag]?
+}
+
+final class CustomFreeTextAnnotationView: FreeTextAnnotationView {
+    var annotationKey: PDFReaderState.AnnotationKey?
+    weak var delegate: FreeTextInputDelegate?
+
+    override func textViewForEditing() -> UITextView {
+        let textView = super.textViewForEditing()
+        if let annotationKey, let delegate {
+            let view = FreeTextInputAccessory(key: annotationKey, delegate: delegate)
+            textView.inputAccessoryView = view
+        }
+        return textView
+    }
+}
+
+final class FreeTextInputAccessory: UIView {
+    private weak var delegate: FreeTextInputDelegate?
+    private let disposeBag: DisposeBag
+
+    init(key: PDFReaderState.AnnotationKey, delegate: FreeTextInputDelegate) {
+        self.delegate = delegate
+        self.disposeBag = DisposeBag()
+        super.init(frame: CGRect(x: 0, y: 0, width: 0, height: 44))
+        self.autoresizingMask = .flexibleWidth
+        self.backgroundColor = .systemBackground
+
+        let separator = UIView()
+        separator.backgroundColor = .opaqueSeparator
+        let separator2 = UIView()
+        separator2.backgroundColor = .opaqueSeparator
+        let separator3 = UIView()
+        separator3.backgroundColor = .opaqueSeparator
+
+        let sizePicker = FontSizeView(contentInsets: UIEdgeInsets.zero)
+        sizePicker.value = delegate.getFontSize(for: key) ?? 0
+        sizePicker.valueObservable
+            .observe(on: MainScheduler.instance)
+            .subscribe(with: self, onNext: { `self`, value in
+                self.delegate?.change(fontSize: value, for: key)
+            })
+            .disposed(by: self.disposeBag)
+        sizePicker.tapObservable
+            .observe(on: MainScheduler.instance)
+            .subscribe(with: self, onNext: { [weak sizePicker] `self`, _ in
+                guard let sizePicker else { return }
+                self.delegate?.showFontSizePicker(sender: sizePicker, key: key, updated: { [weak sizePicker] size in
+                    guard let sizePicker else { return }
+                    sizePicker.value = size
+                })
+            })
+            .disposed(by: self.disposeBag)
+
+        let colorButton = UIButton()
+        let deleteButton = UIButton()
+
+        if #available(iOS 15.0, *) {
```

Comment:
Latest minimum deployment should make this always available.

Reply:
Fixed in e5162c413a6ad621c4e1b0283d49c839b54a4e72